packages/core/src/jsx.js

```diff
@@ -87,15 +98,26 @@ export const jsx: typeof React.createElement = function(
       }\``
     )
   }
-  return consume(context => {
-    // use Context.read for the theme when it's stable
-    if (typeof props.css === 'function') {
-      return (
-        <ThemeContext.Consumer>
-          {theme => render(context, props, type, args, theme)}
-        </ThemeContext.Consumer>
-      )
+
+  let argsLength = args.length
+
+  let createElementArgArray = new Array(argsLength)
```

Comment:
that would be an interesting optimization to make in babel, to postpone argument spreading until they are used 🤔